gradle/crossVersionTest.gradle

```diff
@@ -0,0 +1,110 @@
+import org.gradle.testing.CrossVersionTest
+
+apply plugin: 'java'
+
+sourceSets {
+    crossVersionTest {
+        compileClasspath += main.output
+        runtimeClasspath += main.output
+    }
+}
+
+configurations {
+    crossVersionTestCompile.extendsFrom testCompile
+    crossVersionTestImplementation.extendsFrom testImplementation
+    crossVersionTestRuntime.extendsFrom testRuntime
+    crossVersionTestRuntimeOnly.extendsFrom testRuntimeOnly
+}
+
+dependencies {
+    crossVersionTestCompile project(":internalIntegTesting")
+
+    //so that implicit help tasks are available:
+    crossVersionTestRuntime project(':diagnostics')
+
+    //So that the wrapper and init task are added when crossVersionTests are run via commandline
+    crossVersionTestRuntime project(':buildInit')
+    //above can be removed when we implement the auto-apply plugins
+
+    crossVersionTestRuntime project(":toolingApiBuilders")
+}
+
+useTestFixtures(project: ":toolingApi")
+
+ext.crossVersionTestTasks = tasks.withType(CrossVersionTest)
+
+crossVersionTestTasks.all { CrossVersionTest task ->
+    group = "verification"
+    testClassesDirs = sourceSets.crossVersionTest.output.classesDirs
+    classpath = sourceSets.crossVersionTest.runtimeClasspath
+    requiresBinZip = true
+    requiresLibsRepo = true
```

Comment:
⭕️ I think the equivalent line in `integ-test.gradle` can now be removed, saving a bit of time. Not 100% sure though.